src/components/application_manager/src/commands/request_to_hmi.cc

```diff
@@ -50,13 +51,46 @@ bool ChangeInterfaceState(ApplicationManager& application_manager,
   if (response_from_hmi[strings::msg_params].keyExists(strings::available)) {
     const bool is_available =
         response_from_hmi[strings::msg_params][strings::available].asBool();
-    const HmiInterfaces::InterfaceState interface_state =
-        is_available ? HmiInterfaces::STATE_AVAILABLE
-                     : HmiInterfaces::STATE_NOT_AVAILABLE;
-    application_manager.hmi_interfaces().SetInterfaceState(interface,
-                                                           interface_state);
+
+    if (!is_available) {
+      application_manager.hmi_interfaces().SetInterfaceState(
+          interface, HmiInterfaces::STATE_NOT_AVAILABLE);
+      return false;
+    }
+
+    if (response_from_hmi[strings::params].keyExists(hmi_response::code)) {
```

Comment:
extract code block of parsing error to lambda or function

Reply:
Done in commit 617fdd9